app/controllers/get_involved_controller.rb

```diff
@@ -0,0 +1,94 @@
+class GetInvolvedController < ApplicationController
+
+  attr_accessor :content_item
+  
+  def show
+    load_content_item
+    load_get_involved_data
+    @do_not_show_breadcrumbs = true
+    render template: "content_items/get_involved"
+  end
+
+  def load_content_item
+    content_item = Services.content_store.content_item("/government/get-involved")
+
+    if Services.feature_toggler.use_recommended_related_links?(content_item["links"], request.headers)
+      content_item["links"]["ordered_related_items"] = content_item["links"].fetch("suggested_ordered_related_items", [])
+    end
+
+    @content_item = PresenterBuilder.new(
+      content_item,
+      content_item_path,
+      view_context,
+    ).presenter
+  end
+
+  def load_get_involved_data
+    @open_consultation_count = retrieve_given_document_type("open_consultation")['total']
+    @closed_consultation_count = retrieve_date_filtered_closed_consultations(12)
+    @next_closing_consultations = [retrieve_next_closing] 
+    @recently_opened_consultations = retrieve_new_consultations
+    @recent_consultation_outcomes = retrieve_consultation_outcomes
+    @take_part_pages = sort_take_part(retrieve_given_document_type("take_part")['results'])
+  end
+
+  def retrieve_given_document_type(document_type)
+    Services.publishing_api.get_content_items(document_type: document_type).to_hash
+  end
+
+  def retrieve_special(params)
+    Services.publishing_api.get_content_items(params).to_hash
+  end
+
+  # Aims to find a count of consultations closed in the last n months.
+  # Could feasibly be done via SearchAPI calls but there is no direct or indirect way to search by closing date
+  # there so we have to go around the houses a bit via a single large P-API call.
+  def retrieve_date_filtered_closed_consultations(months)
+    closed_count = 0
+
+    cl_docs = retrieve_special({document_type: 'closed_consultation', per_page: 500, fields: ['details']})['results']
+    cl_docs.sort_by! { |k| -k['details']['closing_date'] }.reverse!
+
+    cl_docs.each do |doc|
+      if DateTime.parse(doc['details']['closing_date']) > DateTime.now.prev_month(months)
+        closed_count += 1
+      else
+        break
+      end
+    end
+
+    return closed_count # hard return for clarity
+  end
+
+  def retrieve_next_closing
+    open_consults = retrieve_given_document_type("open_consultation")['results']
+    open_consults.sort_by! { |k| k['details']['closing_date'] }[0]
+  end
+
+  def retrieve_new_consultations
```

Comment:
Given that this method and the one above it have the same call to `retrieve_given_document_type`, I'd suggest passing that in as an argument instead. It would mean defining what the 'open consults' are in only one place (load_get_involved_data) and make it easier to test, too.

```suggestion
  def retrieve_new_consultations(open_consults)
```